virttest/utils_libvirt/libvirt_network.py

```diff
@@ -78,6 +79,26 @@ def check_established(params):
     port_to_check = params.get("port_to_check", "4915")
     check_local = 'yes' == params.get("check_local_port", "no")
     ipv6_config = "yes" == params.get("ipv6_config", "no")
+    check_network_transport_conn_num = params.get("check_network_transport_conn_num")
+    check_socket_num = "yes" == params.get("check_socket_num", "no")
+    exp_num = params.get("expected_socket_num", "2")
+    sockets = None
+
+    def _check_socket():
+        _res = process.run(cmd, shell=True).stdout_text.split('\n')
+        global sockets
+        sockets = [x.strip() for x in _res if socket_pattern.match(x)]
+        logging.debug("Found sockets: %s", sockets)
+        return len(sockets) == int(exp_num)
+
+    if check_socket_num:
+        cmd = ("netstat -xanp| grep \"CONNECTED\"")
+        socket_pattern = re.compile(r".*(desturi-socket|migrateuri-socket).*")
+        found_expected = utils_misc.wait_for(_check_socket, timeout=30)
+        if not found_expected:
+            raise exceptions.TestFail("There should be {} connected unix sockets, "
+                                      "but found {} sockets.".format(exp_num, len(sockets)))
```

Comment:
In terms of sockets value, you may fail to get its updated value after calling _check_socket although it declare as global variable. But that need take effective out of check_established() function.

```
>>> def check_established():
...     sockets = None
...     def _check_socket():
...         global sockets
...         sockets = [1,2]
...     _check_socket()
...     print(sockets)
... 
>>> check_established()
None

```

Reply:
Thanks, updated.